src/App.js

```diff
@@ -1,8 +1,190 @@
+import {
+  BrowserRouter as Router, NavLink, Route, Routes,
+} from 'react-router-dom';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import { Nav, Modal } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect, useState } from 'react';
+import thunkUser from './Redux/Username/thunk/thunk';
+import thunkLikes from './Redux/Likes/Thunk/thunk';
+import Splash from './components/registeration/Splash';
+import Sign from './components/registeration/Sign';
+import Detail from './components/cars/detail';
+import MyReservations from './components/Reservations/MyReservations';
+import NewReservation from './components/Reservations/NewReserve';
+import NewCar from './components/cars/newcar';
+import fetchAllCars from './Redux/cars/fetch/fetchcars';
+import fetchReserve from './Redux/Reserve/thunk/Fetch_reserve';
+import DOWN from './assets/images/down.png';
+import './app.css';
 
-function App() {
+const App = () => {
+  const dispatch = useDispatch();
+  const [lgShow, setLgShow] = useState(false);
+  const session = useSelector((state) => state.session);
+  const currentUser = useSelector((state) => state.current_user);
+  let userPhoto = 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQyYOyr5Ec8JLXH9PnLVHZ2QKcW43XQs47vnQ&usqp=CAU';
+  if (currentUser.photo) {
+    userPhoto = currentUser.photo;
+  }
+  const adjustSize = () => {
+    const header = document.querySelector('.header');
+    const menu = document.querySelector('.menu-icon');
+    const profile = document.querySelector('.profile-icon');
+    const w = window.innerWidth;
+    if (w >= 1024) {
+      header.style.padding = '1.25% 0';
+      menu.style.display = 'none';
+      header.style.justifyContent = 'flex-end';
+      profile.style.width = '4%';
+    }
+  };
+  const profileMenu = (e) => {
+    let { target } = e;
+    while (target.className !== 'profile-icon') {
+      target = target.parentNode;
+    }
+    let signOut = document.querySelector('.sign-out-wrapper');
+    if (!signOut) {
+      const div = document.createElement('div');
+      div.className = 'sign-out-wrapper';
+      div.innerHTML = `<p class='sign-in-as'> Signed in as </p><p class='user-name'>${currentUser.username}</p><hr/><a href="/" class="sign-out">Sign out</a>`;
+      const wrapper = document.querySelector('.wrapper-app');
+      wrapper.appendChild(div);
+      wrapper.zIndex = 1;
+      div.zIndex = 999;
+    }
+    signOut = document.querySelector('.sign-out-wrapper');
+    signOut.style.display = 'flex';
+    const leaveBtn = document.querySelector('.sign-out');
+    leaveBtn.addEventListener('click', () => {
+      localStorage.clear();
+      dispatch({
+        type: 'LOGOUT',
+      });
+    });
+    document.addEventListener('click', (e) => {
+      if (e.target.closest('.profile-icon')) {
+        return;
+      }
+      if (e.target.closest('.sign-out-wrapper')) {
+        return;
+      }
+      signOut.style.display = 'none';
+    });
+  };
```

Comment:
- [ ] Try not to use regular Javascript if you are already using React. I am sure there's a better way to handle this, so I suggest you to try another approach using React. Maybe creating another component would be ideal here.